### PR TITLE
Qualcomm AI Engine Direct - Fixed the order of the transforms for llama

### PR DESCRIPTION
**Pull request by shewu-quic**
We need to apply r1 r2 before converting linear to conv.

### CODE REVIEW

**Comment by shewu-quic** [comment]
Hi @cccclai,
Sorry to bother you.
I miss place the order of the transform when rebase.

Could you help to review it again?
Thanks a lot.

**Comment by shewu-quic** [comment]
I have a small request. I would like to add a document that explains how to export llama with Qualcomm AI Engine Direct, including steps like downloading spin quant and setting num_sharding. However, I’m not sure where the best place to save this file would be. Could you please advise?

**Review by cccclai — APPROVED** [review]
Looks good

**Comment by cccclai** [comment]
CI needs to be fixed.

File: examples/models/llama2/eval_llama_lib.py [hunk 1]
```diff
@@ -60,7 +62,10 @@ def _model_call(self, inps):
                     pos_tensor = torch.tensor([pos], dtype=torch.int64)
                     logits = self._model(inps[:, pos : pos + 1], pos_tensor)
                     result_logits.append(logits)
-                return torch.cat(result_logits, dim=1)
+                if self._generate_full_logits:
+                    return torch.cat(result_logits, dim=1)
+                else:
+                    return torch.stack(result_logits, dim=1)
```
[the remark this answers is not in the document]
Comment (shewu-quic) [reply examples/models/llama2/eval_llama_lib.py:68]: Because the shape of the function output should be (batch, seq, vocab_size).
If _generate_full_logits, the shape of each result logit in result_logits are (batch, seq, vocab_size)
We could just use cat by dim=1.
If not _generate_full_logits, the shape of each result logit in result_logits are (batch, vocab_size).
We will need use stack to get one more dimension (batch, seq, vocab_size)
